File: BCKEND/Java/BancoFinanceiro/src/Model/Transacao.java
package Model;

import java.util.Date;

enum TipoTransacao {
	SAQUE, DEPÓSITO, PAGAMENTO, TRANSFERÊNCIA
}

public class Transacao {

	private TipoTransacao tipo;
	private Date data;
	private double valor;
	private Cliente clienteTransferencia;
	private String infoPagamento;
	private char tipoValor;

	public Transacao(TipoTransacao tipo, Date data, double valor, 
			Cliente clienteTransferencia, String infoPagamento, char tipoValor) {
		super();
		this.tipo = tipo;
		this.data = data;
		this.valor = valor;
		this.clienteTransferencia = clienteTransferencia;
		this.infoPagamento = infoPagamento;
		this.tipoValor = tipoValor;
	}

	public TipoTransacao getTipo() {
		return tipo;
	}

	public Date getData() {
		return data;
	}

	public double getValor() {
		return valor;
	}

	public Cliente getClienteTransferencia() {
		return clienteTransferencia;
	}
	
	public String getInfoPagamento() {
		return infoPagamento;
	}

	public char getTipoValor() {
		return tipoValor;
	}

	@Override
	public String toString() {
		if (clienteTransferencia != null) {
			return data.getDate() + "/" + (data.getMonth()+1)  + "/" + 
					(data.getYear()+1900) + "\t" + tipo + " - " 
					+ clienteTransferencia.getNome() + "\t" + tipoValor + " R$ "+ valor;			
		} else if(infoPagamento != null) {
			return data.getDate() + "/" + (data.getMonth()+1)  + "/" +  
					(data.getYear()+1900) + "\t" + tipo + " - " 
					+ infoPagamento + "\t" + tipoValor + " R$ "+ valor;
		} else {
			return data.getDate() + "/" + (data.getMonth()+1)  + "/" + 
					(data.getYear()+1900) + "\t" + tipo + "\t" + tipoValor + " R$ "+ valor;
		}
		
	}	

}

File: BCKEND/TypeScript/BancoFinanceiro/src/Model/Transacao.ts
import { Cliente } from "./Cliente";

export enum TipoTransacao {
    DEPOSITO = "Depósito",
    SAQUE = "Saque",
    TRANSFERENCIA = "Transferência",
    PAGAMENTO = "Pagamento"
}

export class Transacao {
    private tipo: TipoTransacao;
    private valor: number;
    private data: Date;
    private clienteTransferencia?: Cliente;
    private tipoValor: string;

    constructor(tipo: TipoTransacao, valor: number, tipoValor: string,  clienteTransferencia?: Cliente) {
        this.tipo = tipo;
        this.valor = valor;
        this.data = new Date();
        this.tipoValor = tipoValor;
        if(clienteTransferencia) this.clienteTransferencia = clienteTransferencia;        
    }

    public getTipo(): TipoTransacao {
        return this.tipo;
    }

    public getValor(): number {
        return this.valor;
    }

    public getData(): Date {
        return this.data;
    }

    public getClienteTransferencia(): Cliente | undefined {
        return this.clienteTransferencia;
    }

    public getTipoValor(): string {
        return this.tipoValor;
    }

    public toString(): string {
        let info = `Tipo: ${this.tipo} | Valor: ${this.tipoValor} ${this.valor.toFixed(2)} | Data: ${this.data.toLocaleString()}`;
        if (this.clienteTransferencia) {
            info += ` | Cliente Transferência: ${this.clienteTransferencia.getNome()}`;
        }
        return info;
    }


}